modules/percolator/src/main/java/org/elasticsearch/percolator/PercolateQuery.java

```diff
@@ -206,14 +228,39 @@ public float score() throws IOException {
                         }
                     };
                 } else {
+                    Scorer scorer = verifiedQueriesQueryWeight.scorer(leafReaderContext);
                     return new BaseScorer(this, approximation, queries, percolatorIndexSearcher) {
 
+                        boolean notExhausted = scorer != null;
+                        DocIdSetIterator iterator = notExhausted ? scorer.iterator() : null;
```

Comment:
can we rename it to something more explicit, like `verifiedDocs` or `verifiedDocsIterator` ?